Use php-cs-fixer/shim instead + full Symfony 8 support

Kocal opened the pull request:
Follows #1125

Jean85 commented:
Because you're forcing Composer's SAT solver to use the beta; it seems that, if it's left with the choice, it prefers the stable tag.

We need Flex and its forced requirement.

Kocal commented:
Added Flex and updated the pinned version of `shivammathur/setup-php`. 
We now requires https://github.com/phpbench/container/pull/12 to be merged/released.

Jean85 commented:
> container 2.2.3 was tagged.

Can you please restart the CI?

> still seems odd that Symfny seems to require a special tool to install it, given a choice I'd rather just `composer require symfony/console $symfonyVersion` in the pipeline. But it's fine to leave it as it is.

Flex does a lot more than that, because it locks down all Symfony components, so it influences indirect dependencies too. In the past I had issues without that because it can mix different major versions, leading to strange bugs in CI.

dantleech commented:
ok, looks like we've uncovered actual incompatibilit(y|ies) now
